test/libponyc/util.cc

```diff
@@ -100,17 +100,18 @@ static const char* const _builtin =
   // - call .values() iterator in a for loop
   // - be a subtype of Seq
   // - call genprim_array_serialise_trace (which expects the three fields)
+  "class ArrayValues[A]\n"
+  "  fun ref has_next(): Bool => false\n"
+  "  fun ref next(): A ? => error\n"
   "class Array[A] is Seq[A]\n"
   "  var _size: USize = 0\n"
   "  var _alloc: USize = 0\n"
   "  var _ptr: Pointer[A] = Pointer[A]\n"
   "  new create(len: USize = 0) => true\n"
   "  fun ref push(value: A) => true\n"
   "  fun apply(index: USize): this->A ? => error\n"
-  "  fun values(): Iterator[A] => object ref\n"
-  "    fun ref has_next(): Bool => false\n"
-  "    fun ref next(): A ? => error\n"
-  "  end\n"
+  "  fun values(): Iterator[this->A]^ =>\n"
+  "    ArrayValues[this->A]\n"
```

Comment:
These fake methods were incorrect, but it was being masked due to the faulty subtyping checks